packages/common/src/services/__tests__/filter.service.spec.ts

```diff
@@ -1611,6 +1611,47 @@ describe('FilterService', () => {
     });
   });
 
+
+  describe('drawFilterTemplate method', () => {
+    let mockColumn1: Column;
+    let mockColumn2: Column;
+    let mockColumn3: Column;
+    let mockArgs1;
+    let mockArgs2;
+    let mockNewFilters: CurrentFilter[];
+
+    beforeEach(() => {
+      gridOptionMock.enableFiltering = true;
+      gridOptionMock.enableTreeData = false;
+      gridOptionMock.backendServiceApi = undefined;
+      mockColumn1 = { id: 'firstName', name: 'firstName', field: 'firstName', filterable: true, filter: { model: Filters.inputText } };
+      mockColumn2 = { id: 'isActive', name: 'isActive', field: 'isActive', type: FieldType.boolean, filterable: true, filter: { model: Filters.select, collection: [{ value: true, label: 'True' }, { value: false, label: 'False' }], } };
+      mockColumn3 = { id: 'name', field: 'name', filterable: true, filter: { model: Filters.inputText }  };
+      mockArgs1 = { grid: gridStub, column: mockColumn1, node: document.getElementById(DOM_ELEMENT_ID) };
+      mockArgs2 = { grid: gridStub, column: mockColumn2, node: document.getElementById(DOM_ELEMENT_ID) };
+      mockNewFilters = [
+        { columnId: 'firstName', searchTerms: ['Jane'], operator: 'StartsWith' },
+        { columnId: 'isActive', searchTerms: [false] }
+      ];
+      sharedService.allColumns = [mockColumn1, mockColumn2, mockColumn3];
+    });
+    it('should Draw DOM Element Filter on custom HTML element',  async () => {
+      service.init(gridStub);
+      service.bindLocalOnFilter(gridStub);
+      gridStub.onHeaderRowCellRendered.notify(mockArgs1 as any, new Slick.EventData(), gridStub);
+      gridStub.onHeaderRowCellRendered.notify(mockArgs2 as any, new Slick.EventData(), gridStub);
+      await service.updateFilters(mockNewFilters);
+
+      const filterContainerElm:HTMLDivElement = document.querySelector('#'+DOM_ELEMENT_ID);
+      const output1 = service.drawFilterTemplate('firstName','#'+DOM_ELEMENT_ID);
+      const output2 = service.drawFilterTemplate('isActive',filterContainerElm);
+      const output3 = service.drawFilterTemplate('selector',filterContainerElm);
+      const output4 = service.drawFilterTemplate('name',filterContainerElm);
+            
+      //expect(output).toBe(output.columnDef.id);
```

Comment:
so I see that you're not actually testing anything (you probably get good coverage because you're calling the function but you're not testing anything though), we'll have to fix that. We can probably query the DOM for that element and expect something to be rendered on that filter element, something like `document.body.querySelector('#'+DOM_ELEMENT_ID);` (it would also be better to use string template literal with backtick ` `` ` which is more readable, instead of string concatenation with +)

Reply:
Exactly, I tried to make the test routine observing the other routines, as I had never done this before, I was only able to execute the call for the coverage test, but I couldn't implement the return analysis routine.
I'm really going to need help with this.

Reply:
just try to get the DOM element from the body with what I wrote in previous paragraph and then at least check that it's truthy, something like this `expect(filterElm).toBeTruthy();`, just try to follow the other tests, copy a test and add whatever you need to it

Reply:
hey @mcallegario so I looked at adding full unit test coverage, here's the lines you can add the following tests 

```ts
expect(filterContainerElm).toBeTruthy();
expect(output1).toBeTruthy();
expect(output2).toBeTruthy();
expect(output4).toBeTruthy();
```
Then I can see that the `output3` with `selector` won't return anything because that is not a valid Column name, so perhaps we can use that for another test. We can start by making it more obvious that it's not a valid column name, let's rename `selector` to `invalid-column` with this line
```ts
const output3 = service.drawFilterTemplate('invalid-column', filterContainerElm);
```
and add test to check for null
```ts
expect(output3).toBeNull();
```
but then we're still missing a test when passing a string as the second argument, so we can do something like this
```ts
const output5 = service.drawFilterTemplate('firstName', '.not-exists');
expect(output5).toBeNull();
```
... and there you go full 100% unit test coverage 😉

Reply:
Ready!
I made some adjustments to the test routine.